src/ui/public/es.js

```diff
@@ -9,38 +9,53 @@ import 'elasticsearch-browser';
 import _ from 'lodash';
 import uiModules from 'ui/modules';
 
-let es; // share the client amongst all apps
+const plugins = [function (Client, config) {
+  // esFactory automatically injects the AngularConnector to the config
+  // https://github.com/elastic/elasticsearch-js/blob/master/src/lib/connectors/angular.js
+  _.class(CustomAngularConnector).inherits(config.connectionClass);
+  function CustomAngularConnector(host, config) {
+    CustomAngularConnector.Super.call(this, host, config);
+
+    this.request = _.wrap(this.request, function (request, params, cb) {
+      if (String(params.method).toUpperCase() === 'GET') {
+        params.query = _.defaults({ _: Date.now() }, params.query);
+      }
+
+      return request.call(this, params, cb);
+    });
+  }
+
+  config.connectionClass = CustomAngularConnector;
+}];
+
+
+// share the clients amongst all apps
+let es;
```

Comment:
@jbudz You could probably do it with an angular factory, yeah.  The important part was having the state itself be stored in the context of the current angular application rather than being stored in the javascript module context.